### PR TITLE
Hw3 databases

### PR DESCRIPTION
**Pull request by Sytnikovakatya**

### CODE REVIEW

File: src/controllers/movieControllers.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        next(err);
+    }
+};
+
+export const getMoviesByGenre = async (req: Request, res: Response) => {
+  const genreName = req.params.genreName;
+  const genre: IGenre | null = await genreService.getGenreByName(genreName);
+ 
+  if (!genre?._id) {
+    return res.status(400).json({ message: `Genre '${genreName}' not found.` });
```
Comment (IhorOsadchyi) [inline src/controllers/movieControllers.ts]: Guess 400 is not great code for this type of error.

File: src/service/movie.service.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+};
+
+export const updateMovie = async (
+  id: string,
+  data: IMovie
+): Promise<MovieDocument | null> => {
+  return MovieModel.findOneAndUpdate({ _id: id }, data, {
+    new: true,
+    runValidators: true
+  }).populate("genre", { _id: 1, name: 1 }, GenreModel);
```
Comment (IhorOsadchyi) [inline src/service/movie.service.ts]: I guess you can pass just empty object {} and it should work.

File: src/service/movie.service.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+};
+
+export const getMoviesByGenre = async (
+  id: number
+): Promise<MovieDocument[]> => {
+  return MovieModel.find({ genre: id}).populate(
+    "genre",
+    { _id: 1, name: 1 },
```
Comment (IhorOsadchyi) [inline src/service/movie.service.ts]: here and in other same cases.

File: src/controllers/movieControllers.ts (hunk 1 above)
Comment (Sytnikovakatya) [reply src/controllers/movieControllers.ts]: I've changed it to 404

File: src/service/movie.service.ts (hunk 2 above)
Comment (Sytnikovakatya) [reply src/service/movie.service.ts]: Thanks. Done

File: src/service/movie.service.ts (hunk 3 above)
Comment (Sytnikovakatya) [reply src/service/movie.service.ts]: Done

File: src/routes/genre.routes.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Router } from 'express';
+import * as genreControllers from '../controllers/genreControllers';
+import * as genreValidationRules from '../middleware/genreValidationRules';
+
+const router = Router();
+
+router.get('/', genreControllers.getAllGenres);
+router.post('/', genreControllers.createGenre, genreValidationRules.genreCreateRules);
+router.patch('/:id', genreControllers.updateGenre, genreValidationRules.genreUpdateRules);
+router.delete('/:id', genreControllers.deleteGenre, genreValidationRules.genreDeleteRules);
```
Comment (IhorOsadchyi) [inline src/routes/genre.routes.ts]: Are you sure the validation middleware should be called after the controller is called? Validation will not be called in most cases, as the controller will return the result before passing control to the next middleware. The primary purpose of validation is to validate input data before processing. Please check this point through your code.

Comment (Sytnikovakatya) [reply src/routes/genre.routes.ts]: Yes, I've changed it

**Review by IhorOsadchyi — APPROVED** [review]

**Pull request merged**